### ISSUE TITLE
Modal perf issues

### ISSUE THREAD

**Issue by joshxg** [openmsupply/openmsupply-client#733]
The modal in particular on detail views is lacking in performance!

Suspicion is it is re-rendering too many times.

**Comment by joshxg** [comment]
Spent a bit of time investigating this.

I found that when opening the modal, the modal is re-rendering a fair few times.. however it didn't really explain it enough for me. Renders are quick and it wasn't dozens of times or anything!

I realised after profiling that the top level router was re-rendering.. which of course meant that almost the entire app was re-rendering.. and that made a lot more sense. 

That's happening because of the `item` input. This is using the `useQueryParams` hook which syncs up the filter state of a list with the URL. For this case, it's not quite what we want, but regardless, that is causing two changes to the URL

Currently there are 9 heavy renders:
![image](https://user-images.githubusercontent.com/35858975/152444404-4378bf95-34fa-4db1-b47c-b640d5293a20.png)

1. The 2nd and 3rd are coming from the root router re-rendering. This requires heavy memoising of components all the way down the tree or a rework of the search params in the URL
2. The 4th, 5th 7th & 8th is the whole page re-rendering which is coming from using the outbound shipment hook which can be avoided.
3. The 6th is just the content area and the modal rendering after fetching some data. The content area should not need to re-render here, so need to look more deeply into that as although stopping the whole page renders above can be done, the content area still re-renders which takes the bulk of the page rendering anyway.

So:
- Memoise all the way down the tree or rework the `useSearchParameters` hook
- Stop the whole page re-rendering from the `useOutbound` hook
- Determine why the `ContentArea` component is re-rendering. It is coming from the hooks used: `useColumns`, `useOutboundRows` or `useIsGrouped`
